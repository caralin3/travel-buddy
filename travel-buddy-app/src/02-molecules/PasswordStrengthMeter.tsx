import React from 'react';
import { Progress } from 'reactstrap';

const MAX_STRENGTH = 20;

export interface PasswordStrengthMeterProps {
  value: number;
}

export const PasswordStrengthMeter: React.FC<PasswordStrengthMeterProps> = ({ value }) => {
  const percentage = (value / MAX_STRENGTH) * 100;
  let color = 'danger';

  if (percentage > 75) {
    color = 'success';
  } else if (percentage > 50) {
    color = 'warning';
  }

  return (
    <div className="d-flex mt-2 w-100 align-items-center">
      <small>
        <em>Strength&nbsp;</em>
      </small>
      <Progress className="w-100" style={{ height: 10 }} value={value} max={MAX_STRENGTH} color={color} />
    </div>
  );
};
