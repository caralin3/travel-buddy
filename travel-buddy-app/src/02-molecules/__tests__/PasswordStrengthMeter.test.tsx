import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { PasswordStrengthMeter } from '../PasswordStrengthMeter';

describe('PasswordStrengthMeter', () => {
  it('should render danger color correctly', () => {
    const wrapper = shallow(<PasswordStrengthMeter value={5} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render success color correctly', () => {
    const wrapper = shallow(<PasswordStrengthMeter value={20} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render warning color correctly', () => {
    const wrapper = shallow(<PasswordStrengthMeter value={11} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
