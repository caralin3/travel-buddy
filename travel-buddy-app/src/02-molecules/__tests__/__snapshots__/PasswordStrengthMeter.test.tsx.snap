// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordStrengthMeter should render danger color correctly 1`] = `
<div
  className="d-flex mt-2 w-100 align-items-center"
>
  <small>
    <em>
      Strength 
    </em>
  </small>
  <Progress
    barStyle={Object {}}
    className="w-100"
    color="danger"
    max={20}
    min={0}
    style={
      Object {
        "height": 10,
      }
    }
    tag="div"
    value={5}
  />
</div>
`;

exports[`PasswordStrengthMeter should render success color correctly 1`] = `
<div
  className="d-flex mt-2 w-100 align-items-center"
>
  <small>
    <em>
      Strength 
    </em>
  </small>
  <Progress
    barStyle={Object {}}
    className="w-100"
    color="success"
    max={20}
    min={0}
    style={
      Object {
        "height": 10,
      }
    }
    tag="div"
    value={20}
  />
</div>
`;

exports[`PasswordStrengthMeter should render warning color correctly 1`] = `
<div
  className="d-flex mt-2 w-100 align-items-center"
>
  <small>
    <em>
      Strength 
    </em>
  </small>
  <Progress
    barStyle={Object {}}
    className="w-100"
    color="warning"
    max={20}
    min={0}
    style={
      Object {
        "height": 10,
      }
    }
    tag="div"
    value={11}
  />
</div>
`;
